src/global/signal_handler.h

```diff
@@ -21,6 +21,7 @@
 #include "acconfig.h"
 
 typedef void (*signal_handler_t)(int);
+typedef void (*signal_siginfo_handler_t)(int, siginfo_t *, void *);
```

Comment:
could you move this `typedef` into the `.cc` file. as nobody is referencing it in this header file or any source file including it.